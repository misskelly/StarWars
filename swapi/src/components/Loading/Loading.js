import React from 'react'

const Loading = () => {
  return (
    <div className='loading-screen'>
    <p className='loading-text'>Loading...</p>
     {/* <iframe src='https://giphy.com/embed/3ohuA9VZ0k8PMjLENa' title='Yoda how embarassing'
    allowFullScreen></iframe> */}
    <iframe src='https://giphy.com/embed/N8Lfh9gWcWYIU' title='Turtle lightsaber battle' className='gif' allowFullScreen></iframe>
    </div>
  )

}

export default Loading;