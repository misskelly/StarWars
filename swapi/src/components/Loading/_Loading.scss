// <div style="width:100%;height:0;padding-bottom:43%;position:relative;"><iframe src="https://giphy.com/embed/3ohuA9VZ0k8PMjLENa"width="100%"height="100%"style="position:absolute"frameBorder="0"class="giphy-embed"allowFullScreen></iframe></div><p><a href="https://giphy.com/gifs/starwars-star-wars-episode-2-3ohuA9VZ0k8PMjLENa">via GIPHY</a></p>

.loading-screen {
  position: absolute;
  display: flex;
  justify-content: center;
  z-index: 20;
  width: 100%;
  flex-direction: column;
  text-align: center;
}

.loading-text {
  font-size: 4rem;
  margin: 3rem;
  padding: 3rem;
}

.gif {
  border: none;
  min-height: 250px;
}