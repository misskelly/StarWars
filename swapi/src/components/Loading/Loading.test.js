import React from 'react';
import Loading from './Loading';
import { shallow } from 'enzyme';



describe('Loading', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Loading />);
  });
  
  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  
});