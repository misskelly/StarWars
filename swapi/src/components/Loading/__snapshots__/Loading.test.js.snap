// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches the snapshot 1`] = `
<div
  className="loading-screen"
>
  <p
    className="loading-text"
  >
    Loading...
  </p>
  <iframe
    className="gif"
    src="https://giphy.com/embed/N8Lfh9gWcWYIU"
    title="Turtle lightsaber battle"
  />
</div>
`;
